specification/servicenetworking/resource-manager/readme.md

```diff
@@ -26,7 +26,18 @@ These are the global settings for the servicenetworking.
 
 ```yaml
 openapi-type: arm
-tag: package-2023-11-01
+tag: package-2024-05-01-preview
+```
+
+### Tag: package-2024-05-01-preview
+
+These settings apply only when `--tag=package-2024-05-01-preview` is specified on the command line.
+
+```yaml $(tag) == '2024-05-01-preview'
```

Comment:
```suggestion
```yaml $(tag) == 'package-2024-05-01-preview'
```

Reply:
@raych1: Thank you for finding this!  How did you know this was the issue?  And do you know why none of the other checks (like LintDiff) would have failed, if `readme.md` were incorrect?

Perhaps we should add a new check to just validate `readme.md` directly, to make errors like this immediately clear?

CC: @konrad-jamrozik

Reply:
I believe LintDiff ran successfully because it ignores the default tag.  I don't see `package-2024-05-01-preview` anywhere in this log:

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3775401&view=logs&j=0574a2a6-2d0a-5ec6-40e4-4c6e2f70bea2&t=80c3e782-49f0-5d1c-70dd-cbee57bdd0c7

Reply:
https://github.com/Azure/azure-rest-api-specs/issues/29065

Reply:
@mikeharder I think it worked in case of LintDiff as openapi-alps passed the package version directly in AutoRest invocation, [line 64](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3775401&view=logs&j=0574a2a6-2d0a-5ec6-40e4-4c6e2f70bea2&t=80c3e782-49f0-5d1c-70dd-cbee57bdd0c7&l=64):

> Executing AutoRest with LintDiff: node /mnt/vss/_work/_tasks/AzureApiValidation_5654d05d-82c1-48da-ad8f-161b817f6d41/0.0.98/private/azure-swagger-validation/azureSwaggerValidation/node_modules/autorest/dist/app.js --v3 --spectral --azure-validator --semantic-validator=false --model-validator=false --message-format=json --openapi-type=arm --openapi-subtype=arm  --use=@microsoft.azure/openapi-validator@2.2.2 --tag=2024-05-01-preview /mnt/vss/_work/1/azure-rest-api-specs/specification/servicenetworking/resource-manager/readme.md

Note this:

> --tag=2024-05-01-preview

I think here Ray deduced this is the problem because of this on [line 1708](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3775401&view=logs&j=0303ead4-2e3b-557b-303e-fe8735d4d18f&t=b29b24f0-8be9-5e66-edce-54fd3565c09c&l=1708):

> 05:13:27.749 cmderr 	[generator automation-v2] error   | [Exception] No input files provided.

I have a mini TSG about this in [azure-openapi-validator CONTRIBUTING.md](https://github.com/Azure/azure-openapi-validator/blob/main/CONTRIBUTING.md#how-to-obtain-pr-lintdiff-check-autorest-command-invocation-details).

And the AutoRest invocation was not passing any `tag` so it used the `default` which is `package-2024-05-01-preview` but it couldn't map it as the file incorrectly had `yaml $(tag) == '2024-05-01-preview'`

> autorest --use=@autorest/go@4.0.0-preview.63 --go --track2 --output-folder=/mnt/vss/_work/1/s/azure-sdk-for-go/src/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/servicenetworking/armservicenetworking --clear-output-folder=false --go.clear-output-folder=false --honor-body-placement=false --remove-unreferenced-types=true ./autorest.md

Reply:
Thanks I understand now, and I will probably catch this more quickly next time.  But I think either azure-sdk-for-go should throw a clearer error message, or this should have been caught by some other check that throws an clear error message.  "Default tag does not exist" should be a top-level clear error.

Reply:
Actually, should autorest be throwing if default tag doesn't exist?  Maybe this is the most direct fix?

Reply:
As discussed elsewhere - it does throw: `error | [Exception] No input files provided.`. It just doesn't seem to be properly processed downstream.

Reply:
https://github.com/Azure/autorest/issues/4965

Reply:
Thanks @mikeharder, that would be great if the message can be improved in autorest.  Another place we validate the `readme.md` is the `Avocado` tool that has multiple rules to validate around the default tag. An example rule is [multiple_default_tags](https://github.com/Azure/avocado?tab=readme-ov-file#multiple_default_tags). 

@konrad-jamrozik I'm not sure if `Avocado` tool is actively used by the spec PR currently as the status in my mind is this check had been removed from the *required* list.